packages/next/src/utils.ts

```diff
@@ -79,6 +80,59 @@ function isDynamicRoute(route: string): boolean {
   return TEST_DYNAMIC_ROUTE.test(route);
 }
 
+/**
+ * Check if a route is a static metadata route and has corresponding source file
+ */
+function getContentTypeFromFile(fileRef: File): string | undefined {
+  if (!fileRef || !('fsPath' in fileRef)) {
+    return undefined;
+  }
+
+  const ext = path.extname(fileRef.fsPath).slice(1);
+  switch (ext) {
+    case 'ico':
+      return 'image/x-icon';
+    case 'png':
+      return 'image/png';
+    case 'jpg':
+    case 'jpeg':
+      return 'image/jpeg';
+    case 'gif':
+      return 'image/gif';
+    case 'svg':
+      return 'image/svg+xml';
+    case 'txt':
+      return 'text/plain';
+    case 'xml':
+      return 'application/xml';
+    case 'json':
+      return 'application/json';
+    case 'webmanifest':
+      return 'application/manifest+json';
+    default:
+      break;
+  }
+  return undefined;
+}
+
+function isSourceFileStaticMetadata(route: string, files: Files): boolean {
+  const pathname = route.replace(/\/route$/, '');
+  const isMetadataPattern = isStaticMetadataRoute(pathname);
+
+  if (isMetadataPattern) {
+    // strip the suffix from pathname
+    const normalizedPathname = pathname.replace(/-\w{6}$/, '');
+    // A set of files in relative paths of source files
+    // app/page.tsx app/icon.svg
+    const filesSet = new Set(Object.keys(files));
+    const hasStaticSourceFile = filesSet.has(`app${normalizedPathname}`);
```

Comment:
Why is this check necessary? Also I don't think we should be checking from `files` and instead could check from `lambdaPageKeys` like done elsewhere if this is necessary.

Reply:
`lambdaPageKeys` seems from the output, the above two cases output is the same but input varies. I think we still need to verify the source files